resallocserver/manager.py

```diff
@@ -298,6 +303,43 @@ def job(self):
         self.event.set()
 
 
+class CleanUnknownWorker(Worker):
+    """
+    Delete all resources that are not recognized by resalloc
+    https://github.com/praiskup/resalloc/issues/88
+    """
+    def job(self):
+        if not self.pool.cmd_list:
+            return
+        resources = self._list_resources()
+
+        # There is no assigned resource ID for the given resource, that's
+        # the whole point. Let's simply use 0, which means the termination
+        # output for all unknown resources will be logged into
+        # hooks/000000_terminate
+        res_id=0
+
+        for resource in resources:
+            run_command(
+                self.pool.id,
+                res_id,
+                resource,
+                None,
+                self.pool.cmd_delete,
+                'terminate',
+                data=None,
+            )
+
+    def _list_resources(self):
+        env = command_env(pool_id=self.pool.id)
+        cmd = [self.pool.cmd_list]
+
+        proc = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+        return out.decode("utf-8").strip().split("\n")
```

Comment:
This command will return the list of _all_ resources, even those that are not to be terminated.

Reply:
Oops :-)
Fixed.